### PR TITLE
Inbound NAT pools PowerShell changes.

### PR DESCRIPTION
**Pull request by ankurchoubeymsft**
PowerShell commands and model changes for configuring inbound NAT pools on the load balancer. Test cases for inbound NAT pools and their corresponding SessionRecords. Updating package version for Microsoft.Azure.Management.Network. Fixed a few minor typos.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ankurchoubeymsft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by DeepakRajendranMsft** [comment]
can we kick off the job again, seems to be a test timeout issue in azure backup
D:\workspace\powershell\AzurePowershell.test.targets(100,5): error : XUnit tests in assembly "Microsoft.Azure.Commands.AzureBackup.Test.dll" failed or timed out. Ensure that all tests in a project pass and collectively take less than 1 minute to run. [D:\workspace\powershell\build.proj]

**Comment by stankovski** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/LoadBalancerTests.ps1 [hunk 1]
```diff
@@ -1117,4 +1117,190 @@ function Test-LoadBalancer-NicAssociationDuringCreate
...
+        $subnet = New-AzureRMVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix 10.0.1.0/24
+        $vnet = New-AzureRMvirtualNetwork -Name $vnetName -ResourceGroupName $rgname -Location $location -AddressPrefix 10.0.0.0/16 -Subnet $subnet
+        
+        # Create the publicip
+        $publicip = New-AzureRMPublicIpAddress -ResourceGroupName $rgname -name $publicIpName -location $location -AllocationMethod Dynamic -DomainNameLabel $domainNameLabel
+
+        $frontend = New-AzureRMLoadBalancerFrontendIpConfig -Name $frontendName -SubnetId $vnet.Subnets[0].Id
+        New-AzureRMLoadBalancer -Name $lbName -ResourceGroupName $rgname -Location $location -FrontendIpConfiguration $frontend 
+        
+        $lb = Get-AzureRMLoadBalancer -Name $lbName -ResourceGroupName $rgname
```
Comment (stankovski) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/LoadBalancerTests.ps1]: Nit: cmdlets should be AzureRmXXX

File: src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/AddAzureLoadBalancerInboundNatPoolConfigCommand.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Add, "AzureRMLoadBalancerInboundNatPoolConfig"), OutputType(typeof(PSLoadBalancer))]
+    public class AddAzureLoadBalancerInboundNatPoolConfigCommand : AzureLoadBalancerInboundNatPoolConfigBase
```
Comment (stankovski) [inline src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/AddAzureLoadBalancerInboundNatPoolConfigCommand.cs]: Rename to AzureRm from AzureRM

File: src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/GetAzureLoadBalancerInboundNatPoolConfigCommand.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRMLoadBalancerInboundNatPoolConfig"), OutputType(typeof(PSInboundNatPool))]
```
Comment (stankovski) [inline src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/GetAzureLoadBalancerInboundNatPoolConfigCommand.cs]: Same as above

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/LoadBalancerTests.ps1 (hunk 1 above)
Comment (ankurchoubeymsft) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/LoadBalancerTests.ps1]: Fixed in the latest commit. (https://github.com/ankurchoubeymsft/azure-powershell/commit/e6cb77ad5e8e7d3875507ab8f08772cd09d8ea83)

File: src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/AddAzureLoadBalancerInboundNatPoolConfigCommand.cs (hunk 2 above)
Comment (ankurchoubeymsft) [inline src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/AddAzureLoadBalancerInboundNatPoolConfigCommand.cs]: Fixed in the latest commit. (https://github.com/ankurchoubeymsft/azure-powershell/commit/e6cb77ad5e8e7d3875507ab8f08772cd09d8ea83)

File: src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/GetAzureLoadBalancerInboundNatPoolConfigCommand.cs (hunk 3 above)
Comment (ankurchoubeymsft) [inline src/ResourceManager/Network/Commands.Network/LoadBalancer/InboundNatPool/GetAzureLoadBalancerInboundNatPoolConfigCommand.cs]: Fixed in the latest commit. (https://github.com/ankurchoubeymsft/azure-powershell/commit/e6cb77ad5e8e7d3875507ab8f08772cd09d8ea83)

**Comment by ankurchoubeymsft** [comment]
Addressed code review comments. New build succeeded and all tests passed. Can one of the admins retest/verify the pull request.

**Pull request merged**